src/engine/SCons/Tool/MSCommon/vc.py

```diff
@@ -299,44 +334,44 @@ def find_vc_pdir_vswhere(msvc_version):
         debug("Unknown version of MSVC: %s" % msvc_version)
         raise UnsupportedVersion("Unknown version %s" % msvc_version)
 
-    # For bug 3333 - support default location of vswhere for both 64 and 32 bit windows
-    # installs.
-    for pf in ['Program Files (x86)', 'Program Files']:
+    # For bug 3333: support default location of vswhere for both
+    # 64 and 32 bit windows installs.
+    # For bug 3542: also accommodate not being on C: drive.
+    pfpaths = [os.environ["ProgramFiles"]]
+    with suppress(KeyError):
```

Comment:
Can you switch to conventional try/except?

Reply:
Of course, it's up to your preference, but this is considered the more modern way; there are even checkers that object if you use try only to skip an expected result (`try: block; except(FooError): pass`)